js/generation.js

```diff
@@ -0,0 +1,104 @@
+import { createMocks } from './create-mocks.js';
+
+const POPUP = document.querySelector('#card').content.querySelector('.popup');
+const GENERATION_MOCKS = createMocks();
+const TYPE_ROOMS = {
+  flat: 'Квартира',
+  bungalow: 'Бунгало',
+  house: 'Дом',
+  palace: 'Дворец',
+  hotel: 'Отель'
+};
+const MAP_CANVAS = document.querySelector('.map__canvas');
+const NEW_FRAGMENT = document.createDocumentFragment();
+
+GENERATION_MOCKS.forEach((mock) => {
+  const COPY_POPUP = POPUP.cloneNode(true);
+  if (mock.offer.title) {
+    COPY_POPUP.querySelector('.popup__title').textContent = mock.offer.title;
+  }
+  else {
+    COPY_POPUP.querySelector('.popup__title').remove();
```

Comment:
тут в каждой ветке else выполняется похожая логика, поэтому её лучше вынести в функцию, например так
```suggestion
    // саму функцию вынести наверх до всех блоков if
      const removeElement = (selector) => {
    COPY_POPUP.querySelector(selector).remove();
  };
  // и потом использовать в блоках else
  removeElement('.popup__title')
```